wire-library/wire-java-generator/src/test/java/com/squareup/wire/java/JavaGeneratorTest.java

```diff
@@ -514,7 +514,7 @@ public void generateTypeUsesPackageNameOnFieldAndClassNameClash() throws Excepti
       .build();
     assertThat(new JavaWithProfilesGenerator(schema)
       .generateJava("common.proto.Person"))
-        .contains("public final Gender common_proto_Gender;");
+        .contains("public final Gender common_proto_Person_Gender;");
```

Comment:
@swankjesse, could you provide some insight on why we were doing conditional use of fully-qualified names?

I think I have to change these expectations back, since these would potentially result in backwards incompatibility issues with users of Wire.

But I'm not clear on _why_ it was behaving the way it was. In general, this is now "fixed" -- in that the code was trying to use a "qualified name", but was not actually using the real qualified name for the field. The condition that does this is in `KotlinGenerator.kt`, [here](https://github.com/square/wire/blob/e259584f901b24214be1841fb86fced38e848e33/wire-library/wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt#L442-L448).

The actual conditions are not clear to me. I _think_ this might be something about naming conflicts, but it's really unclear how the checks in the condition map precisely to name conflicts in the generated code.

So not only am I unclear on the premise of this name-mangling, I am also looking for input as to whether we can call this change a "fix" or we need to do something else in the kotlin generator to preserve the old naming behavior.